Recon/src/Recon/OverlayProducer.cxx

```diff
@@ -358,26 +360,26 @@ void OverlayProducer::onProcessStart() {
   }
 
   // replace by this line once the corresponding tweak to EventFile is ready:
-  //	overlayFile_ = std::make_unique<framework::EventFile>( overlayFileName_,
-  //true );
-  overlayFile_ = std::make_unique<framework::EventFile>(overlayFileName_);
+  overlayFile_ = std::make_unique<framework::EventFile>( overlayFileName_,true );
   overlayFile_->setupEvent(&overlayEvent_);
-
   // we update the iterator at the end of each event. so do this once here to
   // grab the first event in the processor
-  // TODO this could also be done N random times to get a randomness in which
-  // events get matched to what sim event. noticed that shifting by a fair chunk
-  // helps remove some weak but suspicious correlations between sim and overlay
-  // particle positions. leave it hardwired until we can reset the overlay event
-  // counter in nextEvent() (implemented with the EventFile definition above)
-  int nEventsShift_ = 23;
-  for (int iShift = 0; iShift < nEventsShift_; iShift++) {
+
+  TRandom2 * rd = new TRandom2( 0.); //RNSS undefined before produce()
```

Comment:
I would really like us to figure out how you can use RNSS here so that overlay files can be faithfully re-simulated from scratch. Technically, the RNSS is not initialized until the `onNewRun` method of the conditions object provider is called.

https://github.com/LDMX-Software/Framework/blob/1d5c8aac2004444a58a746a52dcfe3b54d98c977/src/Framework/Process.cxx#L163-L170

Notice that _producers_ are called before _and_ after the conditions are initialized so that they can indirectly affect the conditions object providers via the run header. This is actually used by the simulator so that it can tell the EcalGeometryProvider what geometry to use.

For this case, I think you would be able to move the "initialization" of the overlay file to the `onNewRun` method. This is _after_ the RNSS is initialized, so I think you would be able to access it in the normal way. You can look at the Simulator for how it initializes the Geant4 random number generator.

https://github.com/LDMX-Software/SimCore/blob/61c92b0225e83a9f422558daef80f2afb4be8e35/src/SimCore/Simulator.cxx#L283-L291

Reply:
Cool, thanks, I'll give it a try!

Reply:
Alright, done, works